dask/array/core.py

```diff
@@ -1938,14 +1997,85 @@ def normalize_chunks(chunks, shape=None):
             raise ValueError("Input array has %d dimensions but the supplied "
                              "chunks has only %d dimensions" %
                              (len(shape), len(chunks)))
-        if not all(c == s or (math.isnan(c) and math.isnan(s))
+        if not all(c == s or (math.isnan(c) or math.isnan(s))
```

Comment:
Not sure how you get nan here

Reply:
People can ask for some chunks to be unknown.  We use the value np.nan to signify "I don't know the size"